data_iterator.go

```diff
@@ -1,14 +1,26 @@
 package ghostferry
 
 import (
-	"fmt"
 	sql "github.com/Shopify/ghostferry/sqlwrapper"
 	"math"
 	"sync"
 
 	"github.com/sirupsen/logrus"
 )
 
+type GracefulShutdownError struct{}
+
+func (e GracefulShutdownError) Error() string {
+	return "graceful shutdown"
+}
+
+type PaginationKey struct {
+	table              *TableSchema
+	batchIndex         uint64
+	startPaginationKey uint64
+	endPaginationKey   uint64
+}
```

Comment:
This struct is confusingly named for me. Can we just use the `MinMaxKeys` struct already defined (possibly extending that with the `table` variable here), especially since we initialize variables of this type with the MinMaxKey data?

Reply:
Agree. Should find a new name or way to store this.